app/src/main/java/app/wefridge/wefridge/EditFragment.kt

```diff
@@ -17,39 +51,548 @@ import app.wefridge.wefridge.placeholder.PlaceholderContent
+                dialogInterface.run { startActivity(Intent(android.provider.Settings.ACTION_LOCATION_SOURCE_SETTINGS)) }
+            }
+            .show()
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun adaptUIToModel() {
+        hideItemSaveButtonOnExistingModelId()
+        fillFieldsWithModelContent()
+    }
+
+    private fun hideItemSaveButtonOnExistingModelId() {
+        if (!ADD_ITEM_MODE) itemSaveButton?.isVisible = false
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun fillFieldsWithModelContent() {
+        itemNameTextInputLayout.editText?.setText(model?.name)
+        model?.quantity?.toString()?.let { itemQuantityTextInputLayout.editText?.setText(it) }
+        matchUnitValueToUnitDropdownSelection()?.let { unit_dropdown.editText?.setText(it) }
+        model?.bestByDate?.let { setDatePickerDateTo(it); setDateStringToBestByDateEditText() }
+        itemIsSharedSwitch.isChecked = model?.isShared ?: false
+        model?.location?.let { itemAddressTextInputLayout.editText?.setText(buildAddressString(it)) }
+        itemDescriptionTextInputLayout.editText?.setText(model?.description)
+    }
+
+    private fun matchUnitValueToUnitDropdownSelection(): String? {
+        return when (model?.unit?.value) {
+           Unit.GRAM.value -> getString(R.string.itemUnitGramText)
+            Unit.KILOGRAM.value -> getString(R.string.itemUnitKilogramText)
+            Unit.LITER.value -> getString(R.string.itemUnitLiterText)
+            Unit.MILLILITER.value -> getString(R.string.itemUnitMilliliterText)
+            Unit.OUNCE.value -> getString(R.string.itemUnitOunceText)
+            Unit.PIECE.value -> getString(R.string.itemUnitPieceText)
+            else -> null
+        }
+
+    }
+
+    private fun matchUnitDropdownSelectionToUnit(): Unit? {
+        Log.d("EditFragment", unit_dropdown.editText?.text.toString())
+        Log.d("EditFragment", R.string.itemUnitGramText.toString())
+        return when (unit_dropdown.editText?.text.toString()) {
+            getString(R.string.itemUnitGramText) -> Unit.GRAM
+            getString(R.string.itemUnitKilogramText) -> Unit.KILOGRAM
+            getString(R.string.itemUnitLiterText) -> Unit.LITER
+            getString(R.string.itemUnitMilliliterText) -> Unit.MILLILITER
+            getString(R.string.itemUnitOunceText) -> Unit.OUNCE
+            getString(R.string.itemUnitPieceText) -> Unit.PIECE
+            else -> null
+        }
+
+    }
+
+
+
+    private fun setDateStringToBestByDateEditText() {
+        itemBestByDateTextInputLayout.editText?.setText(buildDateStringFromDatePicker())
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun setDatePickerDateTo(date: Date) {
+        val bestByDate = convertToLocalDate(date)
+        itemBestByDatePicker.updateDate(bestByDate.year, bestByDate.monthValue - 1, bestByDate.dayOfMonth)
```

Comment:
Why month -1?

Reply:
The bestByDate's monthValue attribute ranges from 1 to 12, while the DatePicker expects a range from 0 to 11 to properly display the specified date.